### ISSUE TITLE
libpq >= 14 is required since PostgREST v12.2.0

### ISSUE THREAD

**Issue by steve-chavez** [PostgREST/postgrest#3699]
## Problem

PostgREST fails to start on an instance with libpq < 14:

```
"Failed to query the role settings. {"code":"PGRST000","details":"invalid target_session_attrs value: \"read-only\"\n","hint":null,"message":"Database connection error. Retrying the connection."}"
```

(uses the ubuntu-aarch64 release)

This happens because `target_session_attrs` only accepts `read-only` starting from libpq 14. `target_session_attrs` only supports read-write on libp13, see https://www.postgresql.org/docs/13/libpq-connect.html#LIBPQ-CONNECT-TARGET-SESSION-ATTRS.

## Solution

Options:

- Refuse to start on an older libpq and document the requirement.
-  Provide a fallback and not use `read-only` for older libpqs?

**Comment by steve-chavez** [comment]
Wait, this is wrong. We actually set `target_session_attrs=read-write` internally. 

It's the `db-uri` on the docs the one that's set to `read-only`: https://postgrest.org/en/v12/references/listener.html#listener-on-read-replicas

I'll fix this by putting warning on the docs.

**Comment by steve-chavez** [comment]
When this happens PostgREST is in crashloop of:
```
{"code":"PGRST000","details”:"invalid target_session_attrs value: \"read-only\"\n”,"hint”:null,"message”:"Database connection error. Retrying the connection."}
```

Given that, I think we should consider this error as fatal and quit. @laurenceisla WDYT?

**Comment by steve-chavez** [comment]
Added this commit for the docs part https://github.com/PostgREST/postgrest/commit/c9adaed79d48632fd4fa898d51a1e8750d360f55.

Still missing to clear the bug part.

**Comment by laurenceisla** [comment]
> Given that, I think we should consider this error as fatal and quit.

Right, so it's not like we're waiting for the DB to startup... this needs a library update, which can't be recovered automatically (if I'm not mistaken). So yes, I believe it should be considered fatal.
